src/amm/amm.requests.ts

```diff
@@ -40,6 +42,12 @@ export interface PoolPriceRequest extends NetworkSelectionRequest {
   interval: number;
 }
 
+export interface CosmosPoolPriceRequest extends NetworkSelectionRequest {
```

Comment:
Since Gateway is intended to be chain-agnostic, I don't think there should be Cosmos-specific interfaces for requests and responses for fetching pool prices, etc. Instead, please extend and modify the existing interfaces and adapt your code to them. These interfaces should cover how users interact with various exchange types within the AMM family.

Current:
```
export interface PoolPriceRequest extends NetworkSelectionRequest {
  token0: string;
  token1: string;
  fee: string;
  period: number;
  interval: number;
}
export interface CosmosPoolPriceRequest extends NetworkSelectionRequest {
  address: string;
  token0: string;
  token1: string;
}
```

New:
```
export interface PoolPriceRequest extends NetworkSelectionRequest {
  token0: string;
  token1: string;
  address?: string
  fee?: string;
  period?: number;
  interval?: number;
}
```

you can perform downstream validations in the Cosmos code.